products/compute/api.yaml

```diff
@@ -1256,60 +1253,282 @@ objects:
+              that will be used as the hash key for the consistent hash load
+              balancer. If the cookie is not present, it will be generated.
+              This field is applicable if the sessionAffinity is set to HTTP_COOKIE.
+            properties:
+              - !ruby/object:Api::Type::NestedObject
+                name: 'ttl'
+                description: |
+                  Liftime of the cookie.
+                properties:
+                  - !ruby/object:Api::Type::Integer
+                    name: 'seconds'
+                    required: true
+                    description: |
+                      Span of time at a resolution of a second.
+                      Must be from 0 to 315,576,000,000 inclusive.
+                  - !ruby/object:Api::Type::Integer
+                    name: 'nanos'
+                    description: |
+                      Span of time that's a fraction of a second at nanosecond
+                      resolution. Durations less than one second are represented
+                      with a 0 seconds field and a positive nanos field. Must
+                      be from 0 to 999,999,999 inclusive.
+              - !ruby/object:Api::Type::String
+                name: 'name'
+                description: |
+                  Name of the cookie.
+              - !ruby/object:Api::Type::String
+                name: 'path'
+                description: |
+                  Path to set for the cookie.
+          - !ruby/object:Api::Type::String
+            name: 'httpHeaderName'
+            description: |
+              The hash based on the value of the specified header field.
+              This field is applicable if the sessionAffinity is set to HEADER_FIELD.
+          - !ruby/object:Api::Type::Integer
+            name: 'minimumRingSize'
+            default_value: 1024
+            description: |
+              The minimum number of virtual nodes to use for the hash ring.
+              Larger ring sizes result in more granular load
+              distributions. If the number of hosts in the load balancing pool
+              is larger than the ring size, each host will be assigned a single
+              virtual node.
+              Defaults to 1024.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'connectionDraining'
+        description: |
+          Settings for connection draining
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'drainingTimeoutSec'
+            default_value: 300
+            description: |
+              Time for which instance will be drained (not accept new
+              connections, but still work to finish started).
+      - !ruby/object:Api::Type::Time
+        name: 'creationTimestamp'
+        description: |
+          Creation timestamp in RFC3339 text format.
+        output: true
+      - !ruby/object:Api::Type::Array
+        name: 'customRequestHeaders'
```

Comment:
Can you add this to a test?

Reply:
This is actually only supported for EXTERNAL, so I removed it.